stock_picking_auto_create_lot_qty/models/stock_picking.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
+        """
+        Allows to be called either by button or through code
+        """
+        pickings = self.filtered(lambda p: p.picking_type_id.auto_create_lot)
+        lines = pickings.mapped("move_line_ids").filtered(
+            lambda x: (
+                not x.lot_id
+                and not x.lot_name
+                and x.product_id.tracking != "none"
+                and x.product_id.auto_create_lot
+            )
+        )
+        new_lines = pickings.mapped("move_line_ids").filtered(lambda x: (x.qty_done))
```

Comment:
```suggestion
        new_lines = pickings.mapped("move_line_ids").filtered(lambda x: x.qty_done)
```
IMO this improves clarity (at first I thought this was a tuple)

Reply:
done